docs/plugins.md

```diff
@@ -0,0 +1,269 @@
+
+> plugins[] is an array of plugin, where you can define multiple authenticator and authorizer plugins. plugin name should be unique in plugins array.
+
+
+## Plugin Permissions 
+Adhere to below plugin access control to keep your plugin forward compatible.
+* Plugin should read/write file to and from `plugin-home` directory only. Refer [Plugin Installation](#plugin-installation) step2 for `plugin-home` definition
+* Plugin should access port 80 or 443 or port higher than 1024
+
+All other access are forbidden for the plugin.
+
+> Disclaimer: In BETA version your plugin can access any port and can read/write to any location on file system, however you should implement the plugin as per above access permission to keep your plugin compatible with upcoming release of DataHub.
+
+## Migration Of Plugins From application.yml
+If you have any custom Authentication or Authorization plugin define in `authorization` or `authentication` section of  [application.yml](../metadata-service/factories/src/main/resources/application.yml) then migrate them as per below steps.
+
+1. Implement Plugin: For Authentication Plugin follow steps of [Implementing an Authentication Plugin](#implementing-an-authentication-plugin) and for Authorization Plugin follow steps of [Implementing an Authorization Plugin](#implementing-an-authorization-plugin)
+2. Install Plugin: Install the plugins as per steps mentioned in [Plugin Installation](#plugin-installation). Here you need to map the configuration from [application.yml](../metadata-service/factories/src/main/resources/application.yml) to configuration in `config.yml`. This mapping from `application.yml` to `config.yml` is described below 
+
+   **Mapping for Authenticators**
+
+   a. In `config.yml` set `plugins[].type` to `authenticator`
+
+   b. `authentication.authenticators[].type` is mapped to `plugins[].params.className`
+   
+   c. `authentication.authenticators[].configs` is mapped to `plugins[].params.configs`
+   
+   Example Authenticator Plugin configuration in `config.yml`
+
+   ```yaml
+   plugins:
+      - name: "apache-ranger-authenticator"
+         type: "authenticator"
+         enabled: "true"
+         params:
+         className: "com.abc.RangerAuthenticator"
+         configs:
+            username: "foo"
+            password: "fake"
+   
+   ```
+
+   
+   **Mapping for Authorizer**
+
+   a. In `config.yml` set `plugins[].type` to `authorizer`
+
+   b. `authorization.authorizers[].type` is mapped to `plugins[].params.className`
+   
+   c. `authorization.authorizers[].configs` is mapped to `plugins[].params.configs`
+
+   Example Authorizer Plugin configuration in `config.yml`
+
+   ```yaml
+   plugins:
+      - name: "apache-ranger-authorizer"
+         type: "authorizer"
+         enabled: "true"
+         params:
+         className: "com.abc.RangerAuthorizer"
+         configs:
+            username: "foo"
+            password: "fake"
+   
```

Comment:
We should mention here that they should also mount these files into their Docker Containers

Reply:
This information is available in 'Plugin Installation' section and in step 2 I put the reference to [Plugin Installation](#plugin-installation)